xarray/test/test_backends.py

```diff
@@ -1063,6 +1063,35 @@ def test_weakrefs(self):
             self.assertDatasetIdentical(actual, expected)
 
 
+@requires_rasterio
+class TestRasterIO(CFEncodedDataTest, Only32BitTypes, TestCase):
+    def test_write_store(self):
+        # rasterio is read-only for now
+        pass
+
+    def test_orthogonal_indexing(self):
+        # rasterio also does not support list-like indexing
+        pass
+
+    @contextlib.contextmanager
+    def roundtrip(self, data, save_kwargs={}, open_kwargs={}):
+        with create_tmp_file() as tmp_file:
+            data.to_netcdf(tmp_file, engine='scipy', **save_kwargs)
+            with open_dataset(tmp_file, engine='rasterio', **open_kwargs) as ds:
+                yield ds
+
+    def test_weakrefs(self):
+        example = Dataset({'foo': ('x', np.arange(5.0))})
+        expected = example.rename({'foo': 'bar', 'x': 'y'})
+
+        with create_tmp_file() as tmp_file:
+            example.to_netcdf(tmp_file, engine='scipy')
+            on_disk = open_dataset(tmp_file, engine='rasterio')
+            actual = on_disk.rename({'foo': 'bar', 'x': 'y'})
+            del on_disk  # trigger garbage collection
+            self.assertDatasetIdentical(actual, expected)
```

Comment:
@NicWayand - take a look at the PyNio tests.  They are probably the closest analog for what we need to test here.